install.sh

```diff
@@ -0,0 +1,472 @@
+
+  SKIP_ECHO=1
+  sh_c curl \
+    -#fL \
+    -o "$FILE.incomplete" \
+    -C - \
+    "$URL"
+  sh_c mv "$FILE.incomplete" "$FILE"
+  unset SKIP_ECHO
+
+  echo "+ Downloaded into $FILE"
+}
+
+install_macos() {
+  if command_exists brew; then
+    echo "Installing from Homebrew."
+    echo
+
+    sh_c brew install code-server
+
+    return
+  fi
+
+  echo "Homebrew not installed."
+
+  install_archive
+}
+
+install_deb() {
+  echo "Installing v$VERSION deb package from GitHub releases."
+  echo
+
+  fetch "https://github.com/cdr/code-server/releases/download/v$VERSION/code-server_${VERSION}_$ARCH.deb" \
+    "$CACHE_DIR/code-server_${VERSION}_$ARCH.deb"
+  sudo_sh_c dpkg -i "$CACHE_DIR/code-server_${VERSION}_$ARCH.deb"
+
+  echo_systemd_postinstall
+}
+
+install_rpm() {
+  echo "Installing v$VERSION rpm package from GitHub releases."
+  echo
+
+  fetch "https://github.com/cdr/code-server/releases/download/v$VERSION/code-server-$VERSION-$ARCH.rpm" \
+    "$CACHE_DIR/code-server-$VERSION-$ARCH.rpm"
+  sudo_sh_c rpm -i "$CACHE_DIR/code-server-$VERSION-$ARCH.rpm"
+
+  echo_systemd_postinstall
+}
+
+install_aur() {
+  echo "Installing from the AUR."
+  echo
+
+  prev_dir="$PWD"
+  tmp_dir="$(mktemp -d)"
+  cd "$tmp_dir"
+
+  echo "+ Downloading PKGBUILD from https://aur.archlinux.org/cgit/aur.git/snapshot/code-server.tar.gz"
+  SKIP_ECHO=1 sh_c 'curl -fsSL https://aur.archlinux.org/cgit/aur.git/snapshot/code-server.tar.gz | tar -xz --strip-components 1'
+  unset SKIP_ECHO
+  echo "+ Downloaded into $tmp_dir"
+  sh_c makepkg -si
+
+  cd "$prev_dir"
+  rm -Rf "$tmp_dir"
+
+  echo_systemd_postinstall
+}
+
+install_archive() {
+  echo "Installing static release v$VERSION"
+  echo
+
+  fetch "https://github.com/cdr/code-server/releases/download/v$VERSION/code-server-$VERSION-$OS-$ARCH.tar.gz" \
+    "$CACHE_DIR/code-server-$VERSION-$OS-$ARCH.tar.gz"
+
+  sh_c="sh_c"
+  if [ ! -w "$ARCHIVE_INSTALL_PREFIX" ]; then
+    sh_c="sudo_sh_c"
+  fi
+
+  SKIP_ECHO=1 sh_c mkdir -p "$ARCHIVE_INSTALL_PREFIX/lib" "$ARCHIVE_INSTALL_PREFIX/bin"
+  unset SKIP_ECHO
+
+  if [ -e "$ARCHIVE_INSTALL_PREFIX/lib/code-server-$VERSION" ]; then
+    echo
+    echo "code-server-$VERSION is already installed at $ARCHIVE_INSTALL_PREFIX/lib/code-server-$VERSION"
+    echo "Remove it to reinstall."
+    exit 0
+  fi
+  "$sh_c" tar -C "$ARCHIVE_INSTALL_PREFIX/lib" -xzf "$CACHE_DIR/code-server-$VERSION-$OS-$ARCH.tar.gz"
+  "$sh_c" mv -f "$ARCHIVE_INSTALL_PREFIX/lib/code-server-$VERSION-$OS-$ARCH" "$ARCHIVE_INSTALL_PREFIX/lib/code-server-$VERSION"
+  "$sh_c" ln -fs "$ARCHIVE_INSTALL_PREFIX/lib/code-server-$VERSION/bin/code-server" "$ARCHIVE_INSTALL_PREFIX/bin/code-server"
+
+  echo_static_postinstall
+}
+
+install_npm() {
+  if command_exists yarn; then
+    echo "Installing with yarn."
+    echo
+    sh_c yarn global add code-server --unsafe-perm
+    return
+  elif command_exists npm; then
+    echo "Installing with npm."
+    echo
+    sh_c npm install -g code-server --unsafe-perm
```

Comment:
Should we try detecting if we'll need to run this as root? We could check for write access to the directory echoed by `npm config get prefix`.

Yarn has a similar `yarn global bin` but idk if it's worth checking or if it always chooses a writable directory.

Reply:
Mhmm that's true. I forgot npm by default tries to use `/usr/local`. 🤔

Reply:
Fixed.